ui/lib/pki/addon/components/page/pki-issuer-edit.ts

```diff
@@ -0,0 +1,57 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { task } from 'ember-concurrency';
+import { waitFor } from '@ember/test-waiters';
+import errorMessage from 'vault/utils/error-message';
+import RouterService from '@ember/routing/router-service';
+import FlashMessageService from 'vault/services/flash-messages';
+import PkiIssuerModel from 'vault/models/pki/issuer';
+
+interface Args {
+  model: PkiIssuerModel;
+}
+
+export default class PkiIssuerEditComponent extends Component<Args> {
+  @service declare readonly router: RouterService;
+  @service declare readonly flashMessages: FlashMessageService;
+
+  @tracked usageValues: Array<string> = [];
+  @tracked error = null;
+
+  constructor(owner: unknown, args: Args) {
+    super(owner, args);
+    this.usageValues = (this.args.model.usage || '').split(',');
```

Comment:
Would this be a potential use-case for making a transform of attr type `stringArray`? Maybe it doesn't come up enough, but I feel like there are lots cases where we get a string array from the backend and want it to actually be an array of strings

Reply:
Ya that's a really good suggestion. It does seem like this comes up frequently enough that a transform would be helpful so I went ahead and implemented it. I was just about to push but I figured I should run the tests and ran into a catch.

The test for rolling back the attributes on cancel failed and it looks like Ember Data doesn't track the items in an array which is unfortunate. I think in order for it to be worthwhile we would need to figure out a way to retain the rollback functionality. We should definitely keep this in mind because I definitely think it would be valuable but will take a bit more effort than expected.

Reply:
Thank you for investigating!! This is helpful discovery.